src/main/java/io/hops/hopsworks/expat/epipe/EpipeRunner.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * This file is part of Expat
+ * Copyright (C) 2021, Logical Clocks AB. All rights reserved
+ *
+ * Expat is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Expat is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program. If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+package io.hops.hopsworks.expat.epipe;
+
+import io.hops.hopsworks.expat.elastic.ElasticClient;
+import org.apache.http.HttpHost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class EpipeRunner {
+  private static final Logger LOGGER = LogManager.getLogger(EpipeRunner.class);
+  
+  private static final String PROJECTS_CONFIG_LINE = "index = projects";
+  private static final String FEATURESTORE_CONFIG_LINE = "featurestore_index = featurestore";
+  private static final String APP_PROVENANCE_CONFIG_LINE = "app_provenance_index = appprovenance";
+  
+  private static void fixReindexOf(String elasticIndex, List<String> toReindex,
+                                   List<String> configContent, int configContentIdx, String configLine) {
+    if (configContent.get(configContentIdx).startsWith(configLine)) {
+      if(toReindex.contains(elasticIndex)) {
+        toReindex.remove(elasticIndex);
+      } else {
+        //comment out
+        configContent.set(configContentIdx, "#" + configLine);
+      }
+    } else if (configContent.get(configContentIdx).startsWith("#" + configLine)) {
+      if(toReindex.contains(elasticIndex)) {
+        toReindex.remove(elasticIndex);
+        //uncomment
+        configContent.set(configContentIdx, configLine);
+      }
+    }
+  }
+  
+  private static void updateReindexConfig(String reindexConfigPath, List<String> toReindex) throws IOException {
+    Path configFile = Paths.get(reindexConfigPath);
+    List<String> configContent = new ArrayList<>(Files.readAllLines(configFile, StandardCharsets.UTF_8));
+  
+    List<String> toReindexAux = new LinkedList<>(toReindex);
+    for (int i = 0; i < configContent.size(); i++) {
+      fixReindexOf("projects", toReindexAux, configContent, i, PROJECTS_CONFIG_LINE);
```

Comment:
also why to comment out or uncomment these lines?
If you want to control the reindexing these are the variables, either to reindex the projects index or featurestore index or all (both)
https://github.com/logicalclocks/epipe-chef/blob/master/templates/default/config-reindex.ini.erb#L3

Reply:
this code is in chef... this is expat... if the cluster is already running and we want to upgrade it? and also if you run it on the cloud I didn't know if I can rely on the cloud running the chef recipes for upgrades.